x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/pivot/TermsGroupSource.java

```diff
@@ -47,4 +50,9 @@ public Type getType() {
     public static TermsGroupSource fromXContent(final XContentParser parser, boolean lenient) throws IOException {
         return lenient ? LENIENT_PARSER.apply(parser, null) : STRICT_PARSER.apply(parser, null);
     }
+
+    @Override
+    public QueryBuilder getFilterQuery(List<String> changedBuckets) {
+        return new TermsQueryBuilder(field, changedBuckets);
```

Comment:
I did and removed it again, because this depends on `index.max_terms_count` which is default 65k. But the real problem: it's a per-index setting, not cluster-wide. So I would need to get this setting for every index the source matches and ensure we are not over it.

So my thinking right now is to let it fail in search and detect such failures like we do for circuit breakers.

We could also decide for a good default and make it a requirement, meaning: we only support source indexes that have `index.max_terms_count` > threshold. 

I will create an issue to follow up on it.

Reply:
@hendrikmuhs sounds good to me :)